augur/filter.py

```diff
@@ -376,16 +376,27 @@ def run(args):
     if args.subsample_seed:
         random.seed(args.subsample_seed)
     num_excluded_subsamp = 0
-    if args.group_by and (args.sequences_per_group or args.subsample_max_sequences):
+    #if args.group_by and (args.sequences_per_group or args.subsample_max_sequences):
+    if args.subsample_max_sequences or (args.group_by and args.sequences_per_group):
+        
+        #set groups to group_by values
+        if args.group_by:
+            groups = args.group_by
+        #if not specified use dummy category
+        else:
+            groups = ["dummy"]
```

Comment:
I'm sorry for not thinking about this earlier, but we should use an underscore prefix for constructed fields like this just to avoid accidental conflicts with real metadata columns. This is one strategy that pandas uses, for example, to add its own constructed fields.

Can you replace `"dummy"` with `"_dummy"` throughout this PR?